chia/full_node/mempool_manager.py

```diff
@@ -452,6 +465,11 @@ async def validate_spend_bundle(
             # If you reach here it's probably because your program reveal doesn't match the coin's puzzle hash
             return Err.INVALID_SPEND_BUNDLE, None, []
 
+        # Make sure all fast forward spends still have unspent coins
+        for puzzle_hash in fast_forward_puzzle_hashes:
+            if await get_unspent_lineage_info_for_puzzle_hash(puzzle_hash) is None:
+                return Err.DOUBLE_SPEND, None, []
```

Comment:
it's not obvious to me why you defer this check. instead of storing the puzzle hashes to check in `fast_forward_puzzle_hashes` and then checking them, why not just check them directly in the loop above?